third_party/move/move-vm/runtime/src/loader/mod.rs

```diff
@@ -2713,13 +1722,15 @@ impl StructInfo {
 }
 
 pub(crate) struct TypeCache {
-    structs: HashMap<CachedStructIndex, HashMap<Vec<Type>, StructInfo>>,
+    structs: HashMap<StructName, HashMap<Vec<Type>, StructInfo>>,
```

Comment:
This StructInfo should probably be StructCache or smt

Reply:
Did we also try TreeMap here? to compare the performances.

Seems like ppl here really prefer HashMaps (I guess due to it's O(1)?) But it's always a good idea to measure performances when deciding (Unless, of course, one is not possible to be used).

Also, are there any maps like flat_map in c++ that are optimized for performances on modern hardware.